### ISSUE TITLE
Create a `.prefectignore` file when creating a deployment through Python

### ISSUE THREAD

**Issue by khuyentran1401** [PrefectHQ/prefect#7207]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar request and didn't find it.
- [X] I searched the Prefect documentation for this feature.

### Prefect Version

2.x

### Describe the current behavior

The file `.prefectignore` is automatically created when creating a deployment through CLI if it doesn't exist. However, when creating a deployment through the `Deployment` object, the file is not automatically created. 

### Describe the proposed behavior

The `.prefectignore` file should be automatically created when creating a deployment though the `Deployment` object if it doesn't exist. 

### Example Use

An example of my deployment file:

```python
from test2 import test_flow

from prefect.deployments import Deployment

deployment = Deployment.build_from_flow(
    flow=test_flow,
    name="test-flow",
)

deployment.apply()
```

**Comment by bunchesofdonald** [comment]
I hesitate to add a side effect that relies on filesystem IO to initialization of Deployment class. But maybe a method to create a one could make sense.

**Comment by madkinsz** [comment]
I think we do perform a bunch of IO on `build_from_flow`, but there have been complaints about that already https://github.com/PrefectHQ/prefect/issues/7088

**Comment by Trymzet** [comment]
+1 on this, it's a really strange workflow where you have to randomly create an empty `.prefectignore` file before creating a deployment

**Comment by madkinsz** [comment]
If you don't have one we fail to create the deployment? We definitely should not to that. We could also just use our default one in-memory without writing to disk unless there is one on disk then we'd use that.

**Comment by Trymzet** [comment]
Yes, for example, if you have a flow and deployment in one file, it will not run, because Prefect will be looking for `.prefectignore` in the flow run execution env.

You need to put the flow and the deployment in separate files, create `.prefectignore` by hand in local env, then create & apply the deployment.

**Comment by madkinsz** [comment]
I'd welcome a fix to that!

**Comment by jbbqqf** [comment]
Hi! Triaging older issues — I think this one can be closed as out-of-scope for current Prefect.

- The request is specifically about the Prefect 2.x `Deployment(...)` Python object auto-creating `.prefectignore`. That deployment surface no longer exists in Prefect 3.x — deployments are now configured via `prefect.yaml` + `prefect deploy` (see [v3 deployments docs](https://docs.prefect.io/v3/concepts/deployments)). The Prefect 1/2 deploy entry points are migration stubs only: [`src/prefect/agent.py`](https://github.com/PrefectHQ/prefect/blob/main/src/prefect/agent.py#L1-L3).
- In the 3.x flow, `.prefectignore` handling has shifted entirely into the `prefect deploy` CLI, so the original "Deployment object should also create it" ask is no longer well-defined.

Would you mind closing this out?

(Disclosure: I drafted this comment with help from Claude Code while triaging stale issues; the file/line and docs references above were verified manually against the current `main` branch.)
